### ISSUE TITLE
Plan OnionServices V2 deprecation in compatibility with Tor timeline

### ISSUE THREAD

**Issue by evilaliv3** [globaleaks/GlobaLeaks#2894]
On the 2nd of July Tor has officially published their OnionSevice V2 deprecation timeline:  https://blog.torproject.org/v2-deprecation-timeline:
```
1) September 15th, 2020
0.4.4.x: Tor will start warning onion service operators and clients that v2 is deprecated and will be obsolete in version 0.4.6.
 
2) July 15th, 2021
0.4.6.x: Tor will no longer support v2 and support will be removed from the code base.
 
3) October 15th, 2021
We will release new Tor client stable versions for all supported series that will disable v2.
```

I suggest that based on this timeline we could do the following:
```

1)
4.0.X: GlobaLeaks by default supports onion services Version 3 and implements Version 2 for old setups.

2)
4.1.0: GlobaLeaks will automatically bump every onion service to Version 3 and maintain an automatic redirect from Version 2 to Version 3

3) By October 15th, 2021:
GlobaLeaks will remove v2 support from the codebase
```

**Comment by evilaliv3** [comment]
@emartinborregon @martinszy @Stephane-FILTRALA @SylvainFPU: Starting from GlobaLeaks version 4.1.0 each onion service version 2 will be automatically updated by the software to onion service version 3 and globaleaks will automatically implement a redirects for users that will be trying the old onion service.

\cc @asn-d6

**Comment by SylvainFPU** [comment]
Do you have a rough release date for 4.1.0 @evilaliv3 ?

**Comment by evilaliv3** [comment]
@SylvainFPU: I think it will be released by this or the next week max.

**Comment by evilaliv3** [comment]
This will be shipped by today with release 4.1.0
